Add support to customize thank you email template

SudharakaP opened the pull request:
Related to https://github.com/opencollective/opencollective/issues/2682

Related front-end PR: https://github.com/opencollective/opencollective-frontend/pull/7932

znarf commented on migrations/20220622220001-add-custom-thank-you-email-message.js:
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.addColumn('Collectives', 'customThankYouMessage', {
+      type: Sequelize.TEXT,
+    });
+  },
```
Can you just put this in the Collective settings for now?

SudharakaP replied:
We are now storing this in settings. 👍🏽

znarf commented on server/lib/payments.js:
```diff
@@ -612,6 +612,7 @@ const sendOrderConfirmedEmail = async (order, transaction) => {
   const { collective, tier, interval, fromCollective, paymentMethod } = order;
   const user = await order.getUserForActivity();
   const host = await collective.getHostCollective();
+  const parentCollective = await order.collective.getParentCollective();
```
```suggestion
  const parentCollective = await collective.getParentCollective();
```

znarf commented on server/lib/payments.js:
```diff
@@ -641,6 +642,8 @@ const sendOrderConfirmedEmail = async (order, transaction) => {
       firstPayment: true,
       subscriptionsLink: interval && getEditRecurringContributionsUrl(fromCollective),
+      customMessage:
+        order.collective.settings?.thankYouEmailMessage || parentCollective?.settings?.thankYouEmailMessage,
```
```suggestion
        collective.settings?.thankYouEmailMessage || parentCollective?.settings?.thankYouEmailMessage,
```

znarf commented on server/lib/recurring-contributions.js:
```diff
@@ -387,6 +387,7 @@ export async function sendThankYouEmail(order, transaction, isFirstPayment = fal
   const relatedCollectives = await order.collective.getRelatedCollectives(3, 0);
   const user = await order.getUserForActivity();
   const host = await order.collective.getHostCollective();
+  const parentCollective = await order.collective.getParentCollective();
```
```suggestion
  const parentCollective = await collective.getParentCollective();
```

znarf commented on server/lib/recurring-contributions.js:
```diff
@@ -400,6 +401,7 @@ export async function sendThankYouEmail(order, transaction, isFirstPayment = fal
     config: { host: config.host },
     interval: order.Subscription?.interval || order.interval,
     subscriptionsLink: getEditRecurringContributionsUrl(order.fromCollective),
+    customMessage: order.collective.settings?.thankYouEmailMessage || parentCollective?.settings?.thankYouEmailMessage,
```
```suggestion
    customMessage: collective.settings?.thankYouEmailMessage || parentCollective?.settings?.thankYouEmailMessage,
```

znarf commented on templates/emails/thankyou.hbs:
```diff
@@ -27,6 +27,8 @@ Subject: Thank you for your contribution to {{collective.name}}
   <a href="https://opencollective.com/{{collective.slug}}">https://opencollective.com/{{collective.slug}}</a>, and share the link to help us spread the word!
 </p>
 
+{{{customMessage}}}
```
What if there is none?

SudharakaP replied:
When there's is none (the user empties the custom email section in settings or it's a new collective and they haven't configured the custom email section), we don't see anything here, since this {{{customMessage}}} will be empty. Since everything in this template is html, there's no additional spaces or anything added.

znarf reviewed (changes requested):
Looks good. Couple of details to check.

SudharakaP commented:
> Looks good. Couple of details to check.

@znarf: I've addressed the reviews. Let me know if there's any other issues. Will merge tomorrow otherwise.

znarf reviewed (approved):
Looks good.

znarf commented:
Note that there is still no way to differentiate the "custom message" from the generic template. We discussed this and we can't consider the feature shipped without it.

SudharakaP commented:
> Note that there is still no way to differentiate the "custom message" from the generic template. We discussed this and we can't consider the feature shipped without it.

@znarf : Looking at the design again, we have the string `A message from <collective-name>:` above the custom message.  

![image](https://user-images.githubusercontent.com/12435965/178334328-1a85f6e1-431f-4cbc-8a51-fe56f341024d.png)

I've included this and it looks like,

![image](https://user-images.githubusercontent.com/12435965/178334483-597f739a-c4f7-4700-a020-be5e95123850.png)
